build.gradle.kts

```diff
@@ -225,12 +227,26 @@ tasks {
 	}
 
 	javadoc {
-		// of all the javadoc checks (accessibility, html, missing, reference, syntax; see
-		// https://docs.oracle.com/javase/8/docs/technotes/tools/unix/javadoc.html#BEJEFABE)
-		// disable the warning for missing comments and tags because they spam the output
-		// (it does often not make sense to comment every tag; e.g. the @return tag on annotations)
-		(options as CoreJavadocOptions).addStringOption("Xdoclint:accessibility,html,syntax,reference", "-quiet")
-		options.encoding = "UTF-8"
+		// Exclude internal implementation package from javadoc
+		exclude("org/junitpioneer/internal")
+
+		options {
+			// Cast to standard doclet options, see https://github.com/gradle/gradle/issues/7038#issuecomment-448294937
+			this as StandardJavadocDocletOptions
+
+			links = listOf("https://junit.org/junit5/docs/current/api/")
+
+			// Set javadoc `--release` flag (affects which warnings and errors are reported)
+			addStringOption("-release", targetJavaVersion.getMajorVersion())
+
+			// Enable doclint, but ignore warnings for missing tags, see
+			// https://docs.oracle.com/en/java/javase/17/docs/specs/man/javadoc.html#additional-options-provided-by-the-standard-doclet
+			// The Gradle option methods are rather misleading, but a boolean `true` value just makes sure the flag
+			// is passed to javadoc, see https://github.com/gradle/gradle/issues/2354
+			addBooleanOption("Xdoclint:all,-missing", true)
```

Comment:
I wonder if the same is true here... Is this `--missing` or just `-missing`?

Reply:
It is `-missing`. That is part of the `-Xdoclint` option, it is not a standalone option.
See https://docs.oracle.com/en/java/javase/17/docs/specs/man/javadoc.html#additional-options-provided-by-the-standard-doclet (also linked in the comment in the source above)

Reply:
Also feel free to test the changes locally. Only using `Xdoclint:all` will show the undesired warnings about missing javadoc tags (e.g. for parameters). With `-missing` it only shows the other warnings, currently that the code does not use modules but links against the JUnit doc which uses modules, that can be ignored and could also be solved in the future with #567.